examples/glTFViewer/glTFViewer.cpp

```diff
@@ -271,15 +271,13 @@ namespace {
         }
         auto inputState = builder.GetResult();
 
-        auto bindGroupLayoutBuilder = device.CreateBindGroupLayoutBuilder()
-            .SetBindingsType(nxt::ShaderStageBit::Vertex, nxt::BindingType::UniformBuffer, 0, 1)
-            .Clone();
-        if (hasTexture) {
-            bindGroupLayoutBuilder
-                .SetBindingsType(nxt::ShaderStageBit::Fragment, nxt::BindingType::Sampler, 1, 1)
-                .SetBindingsType(nxt::ShaderStageBit::Fragment, nxt::BindingType::SampledTexture, 2, 1);
-        }
-        auto bindGroupLayout = bindGroupLayoutBuilder.GetResult();
+        nxt::BindGroupLayout bindGroupLayout = utils::MakeBindGroupLayout(
+            device, {
+                        {0, nxt::ShaderStageBit::Vertex, nxt::BindingType::UniformBuffer, 1},
+                        {1, nxt::ShaderStageBit::Fragment, nxt::BindingType::Sampler, 1},
+                        {2, nxt::ShaderStageBit::Fragment, nxt::BindingType::SampledTexture,
+                         static_cast<uint32_t>(hasTexture ? 1 : 0)},
```

Comment:
This will fail validation currently

Reply:
MakeBindGroupLayout - the helper function - special-cases the '0' arraySize, and doesn't record it into the BGL descriptor.